### PR TITLE
siril: update to 1.0.2.

### PR DESCRIPTION
**Pull request by andkem**

### CODE REVIEW

**Comment by andkem** [comment]
This fails cross-compilation because you're supposed to compile the bundled librtprocess separately when you cross-compile.

Is it okay to mark this as nocross or do I need to look at packaging librtprocess separately?

**Comment by ericonr** [comment]
I'm not sure this is stable: https://www.siril.org/2020/08/13/siril-1-0-0beta/

But it's probably worth it to get the packaging going c:

Regarding `librtprocess`, given this from the build log:

```
configure: librtprocess not found
checking for cmake... /usr/bin/cmake
configure: librtprocess will be built statically
```

I'd say to either find some way to build it for cross as well, or make a package for it, which I find preferrable.

**Comment by andkem** [comment]
They first released 0.99.4 as a beta test for version 1.0.0, but then they decided that it was stable enough and released 0.99.4 as a proper release.

They bundle the librtprocess code with Siril and build it automatically if it isn't found when you're not cross-compiling. It lives in a subdirectory and uses cmake to build.

I'll have a look at creating a proper package for librtprocess, which I think is preferable to patching the Siril build process to allow building it for cross compilation.

**Comment by andkem** [comment]
I could add that I've been using this version myself since the original 0.99.4 release announcement and haven't seen any stability problems or major issues myself. In fact, it has been working a lot better than the 0.9.12 release.

**Comment by andkem** [comment]
0.99.6 was released. It is still a 1.0 beta although they consider it better than 0.9.12.

**Comment by andkem** [comment]
Version 0.99.8.1 was released. Keeping MR up to date.

File: srcpkgs/wcslib/template [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+configure_args=" --without-pgplot "
+makedepends="cfitsio-devel"
+short_desc="Tools for WCS in FITS headers"
+maintainer="Andreas Kempe <kempe@lysator.liu.se>"
+license="GPL-3.0-or-later"
+homepage="https://www.atnf.csiro.au/people/mcalabre/WCS/"
+distfiles="ftp://ftp.atnf.csiro.au/pub/software/${pkgname}/${pkgname}-${version}.tar.bz2"
+checksum=a21c964136055ad1d68ca9d3e9714374a9dc3ff2f7072622af08cc4f5a0c511c
+
+if [ "$XBPS_CROSS_BUILD" ]; then
```
Comment (ericonr) [inline srcpkgs/wcslib/template:15]:
```suggestion
if [ "$CROSS_BUILD" ]; then
```

File: srcpkgs/siril/template [hunk 2]
```diff
@@ -24,3 +31,8 @@ CFLAGS="-fcommon"
 pre_configure() {
 	NOCONFIGURE=1 ./autogen.sh
 }
+
+# make check doesn't really do anything and is broken.
+do_check() {
+	:
+}
```
Comment (ericonr) [inline srcpkgs/siril/template:38]: Use the new `make_check=no`

File: srcpkgs/wcslib/template [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	configure_args+=" --with-cfitsioinc=${XBPS_CROSS_BASE}/usr/include --with-cfitsiolib=${XBPS_CROSS_BASE}/usr/lib "
+fi
+
+wcslib-devel_package() {
+	short_desc="Tools for WCS in FITS headers - development files"
```
Comment (ericonr) [inline srcpkgs/wcslib/template:20]:
```suggestion
	short_desc+=" - development files"
```

Comment (andkem) [reply srcpkgs/wcslib/template]: Fixed

File: srcpkgs/siril/template (hunk 2 above)
Comment (andkem) [reply srcpkgs/siril/template]: Fixed

File: srcpkgs/wcslib/template (hunk 1 above)
Comment (andkem) [reply srcpkgs/wcslib/template]: Fixed

**Comment by andkem** [comment]
Keeping this PR compiling by adding a few patches I'm looking at upstreaming and adding a missing NDEBUG to librtprocess.

**Comment by andkem** [comment]
Siril 0.99.10 was released. Keeping PR up to date.

**Comment by andkem** [comment]
Siril 1.0.0 was finally released. Since the 0.9.12 release has been considered deprecated for a few years now, I would love it if we could finally get the Siril package updated.

**Comment by andkem** [comment]
Bumped Siril to 1.0.2 to keep the PR up to date.

**Pull request closed without merging**